src/profile-logic/marker-data.js

```diff
@@ -1159,6 +1160,18 @@ export function isFileIoMarker(marker: Marker): boolean {
   return !!(marker.data && marker.data.type === 'FileIO');
 }
 
+/**
+ * Returns true if the marker is an on-thread FileIO marker.
+ */
+export function isOnThreadFileIoMarker(marker: Marker): boolean {
+  return !!(
+    marker.data &&
+    marker.data.type === 'FileIO' &&
+    // If thread ID isn't there, that means it's a FileIO marker belongs to that thread.
+    typeof marker.data.threadId === 'undefined'
```

Comment:
All fileio markers that were present before Gerald's gecko patch were of this type?

Reply:
All existing profiles don't have data.threadId, but in these the only FileIOs that were recorded were from the main thread into the main thread track. So this change is backward-compatible, no need for a converter. And it's space efficient, because data.threadId is only present when needed.